<launch>	  

  <rosparam file="$(find foilboat_controller)/config/config_simulation.yaml" command="load" />

  <node pkg="foilboat_controller" type="controller_node" name="foilboat_controller_node" output="screen" />
  <node pkg="foilboat_controller" type="range_sensor_wrapper_node" name="range_sensor_wrapper_node" output="screen"/>

  <arg name="model" value="$(find foilboat_description)/urdf/planesim_model.xacro"/>

  <param name="robot_description" command="$(find xacro)/xacro $(arg model)" />

  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

  <node pkg="robot_localization" type="ekf_localization_node" name="foilboat_odom">
    <param name="frequency" value="100" />
    <param name="sensor_timeout" value="0.2" />
    <param name="two_d_mode" value="false" />
    <param name="publish_tf" value="true" />
    <param name="publish_acceleration" value="true" />

    <param name="map_frame" value="map" />
    <param name="odom_frame" value="odom" />
    <param name="base_link_frame" value="base_link" />
    <param name="world_frame" value="odom" />

    <param name="imu0" value="/plane_ros/imu" />
    <param name="imu0_differential" value="false"/>
    <param name="imu0_remove_gravitational_acceleration" value="true" />
    <rosparam param="imu0_config">
      [false, false, false,
      true, true, true,
      false, false, false,
      true, true, true,
      false, false, false]
    </rosparam>

    <remap from="odometry/filtered" to="/plane_ros/odometry/filtered/odom" />
  </node>

  <!-- <node pkg="robot_localization" type="navsat_transform_node" name="navsat_transform">
    <param name="frequency" value="10" />

    <remap from="odometry/filtered" to="/plane_ros/odometry/filtered/odom" />
    <remap from="imu/data" to="/plane_ros/imu" />
    <remap from="gps/fix" to="/plane_ros/gps/fix" />
    <remap from="odometry/gps" to="/plane_ros/odometry/gps" />
  </node> -->

  <!-- <node pkg="robot_localization" type="ekf_localization_node"
    name="foilboat_map">
    <param name="frequency" value="10." />
    <param name="sensor_timeout" value="0.2" />
    <param name="two_d_mode" value="false" />
    <param name="publish_tf" value="false" />
    <param name="map_frame" value="map" />
    <param name="odom_frame" value="odom" />
    <param name="base_link_frame" value="base_link" />
    <param name="world_frame" value="map" />

    <param name="imu0" value="/plane_ros/imu" />
    <param name="imu0_differential" value="false"/>
    <param name="imu0_remove_gravitational_acceleration" value="true" />
    <rosparam param="imu0_config">
      [false, false, false,
      true, true, true,
      false, false, false,
      true, true, true,
      true, true, true]
    </rosparam>

    <param name="height0" value="/plane_ros/z_estimate" />
    <param name="height0_differential" value="false"/>
    <rosparam param="height0_config">
      [false, false, true,
      false, false, false,
      false, false, false,
      false, false, false,
      false, false, false]
    </rosparam>

    <param name="gps0" value="/plane_ros/odometry/gps" />
    <param name="gps0_differential" value="false" />
    <rosparam param="gps0_config">
      [true, true, true,
      true, true, true,
      true, true, true,
      true, true, true,
      false, false, false]
    </rosparam>    

    <remap from="odometry/filtered" to="odometry/filtered_map" />
  </node> -->


  
</launch>